content/blog/2020-07-03-optimizing-dvc-remotes.md

```diff
@@ -0,0 +1,390 @@
+switch to querying each file in our dataset individually. If DVC reaches the end
+of the subdirectory without the estimated size going over the threshold, it will
+continue to fetch the full listing for the rest of the remote.
+
+By estimating remote size in DVC 1.0, we can ensure that we always use the
+optimal method when querying remote status.
+
+_Note: The DVC remote structure also allows us to parallelize the second query
+method described earlier. By fetching the contents of each subdirectory
+individually, we can run up to 256 sets of sequential queries at once._
+
+## How DVC 1.0 reduces the number of files to query
+
+A common DVC use case is
+[versioning](/doc/use-cases/versioning-data-and-model-files) the contents of a
+large directory. As the contents of the directory changes over time, DVC will be
+used to push each updated version of the directory into cloud storage. In many
+cases, only a small number of files within that directory will be modified
+between project iterations.
+
+So after the first version of a project is pushed into cloud storage, for
+subsequent versions, only the small subset of changed files actually needs to be
+synchronized with cloud storage. However, prior to 1.0, DVC would always need to
+query for every file in that directory, regardless of whether or not a given
+file had changed since the last time it was pushed to a remote.
+
+Consider a case where a user has an existing directory with 1 million files
+which has been versioned and pushed to a remote with DVC. In the next iteration
+of the project, only a single file in the directory has been modified. We can
+obviously see that everything other than the one modified file will already
+exist in cloud storage. In older versions of DVC, when the user runs `dvc push`
+to synchronize the directory with remote storage, DVC would need to query
+whether or not all 1 million files existed in the remote.
+
+As we established earlier:
+
+> In simple terms, by reducing the size of our query set as much as possible, we
+> can also improve performance.
+
+In DVC 1.0, we are now able to reduce the size of our query set by keeping an
+index of directories which have already been versioned and pushed into remote
+storage. By referencing this index, DVC will "remember" which files already
+exist in a remote, and will remove them from our query set at the start of a
+data sync operation (before we choose a query method, and before we make any
+cloud storage API requests). Using the previous example, DVC 1.0 will only query
+for the single new file, rather than the entire contents of the directory.
+
+_Note: This optimization only applies to DVC versioned directories. Individually
+versioned files (including those added with `dvc add -R`) are not indexed in DVC
+1.0, and will always be queried during remote operations._
+
+## Conclusion
+
+By selecting the optimal status query method, and optimizing the queries
```

Comment:
Can we make it more general - `By utilizing content-addressable way of storing + ... DVC makes sync blazing fast` ...?